crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -433,7 +434,10 @@ pub struct AuthorityEpochTables {
     ///
     /// REQUIRED: all authorities must assign the same shared object versions for each transaction.
     assigned_shared_object_versions_v2: DBMap<TransactionKey, Vec<(ObjectID, SequenceNumber)>>,
+    assigned_shared_object_versions_v3:
+        DBMap<TransactionKey, Vec<((ObjectID, SequenceNumber), SequenceNumber)>>,
```

Comment:
i think its probably worth making a struct for the `(ObjectID, SequenceNumber)` pair so that it is clear that the sequence number here is an initialSharedVersion